go-controller/pkg/factory/factory.go

```diff
@@ -464,17 +462,17 @@ var (
 )
 
 // NewWatchFactory initializes a new watch factory
-func NewWatchFactory(c kubernetes.Interface, eip egressipclientset.Interface, ec egressfirewallclientset.Interface, crd apiextensionsclientset.Interface) (*WatchFactory, error) {
+func NewWatchFactory(ovnClientset *util.OVNClientset) (*WatchFactory, error) {
 	// resync time is 12 hours, none of the resources being watched in ovn-kubernetes have
 	// any race condition where a resync may be required e.g. cni executable on node watching for
 	// events on pods and assuming that an 'ADD' event will contain the annotations put in by
 	// ovnkube master (currently, it is just a 'get' loop)
 	// the downside of making it tight (like 10 minutes) is needless spinning on all resources
 	wf := &WatchFactory{
-		iFactory:    informerfactory.NewSharedInformerFactory(c, resyncInterval),
-		eipFactory:  egressipinformerfactory.NewSharedInformerFactory(eip, resyncInterval),
-		efClientset: ec,
-		crdFactory:  apiextensionsinformerfactory.NewSharedInformerFactory(crd, resyncInterval),
+		iFactory:    informerfactory.NewSharedInformerFactory(ovnClientset.KubeClient, resyncInterval),
+		eipFactory:  egressipinformerfactory.NewSharedInformerFactory(ovnClientset.EgressIPClient, resyncInterval),
+		efClientset: ovnClientset.EgressFirewallClient,
```

Comment:
The egressfirewall code here is gratuitously different from the other code. Can you fix that too? (As a separate commit.)

Reply:
I noticed that too, but I was unsure if its meant to be like this given that we have a different "instantiation procedure" for egress firewall, depending on the CRD watcher. @JacobTanenbaum , any input on this? I could just instantiate it to: `efClientset: egressfirewallinformerfactory.NewSharedInformerFactory(ovnClientset.EgressFirewallClient, resyncInterval)` and have that be overwritten in `InitializeEgressFirewallWatchFactory`. But should I?

Reply:
Scratch my previous comment. That does not apply as this method instantiates the `efClientset` , not `efFactory`. So we need to keep it as it is for now.